backend/api/EventHandlers/MqttEventHandler.cs

```diff
@@ -245,5 +247,38 @@ private async void OnBatteryUpdate(object? sender, MqttReceivedArgs mqttArgs)
                 _logger.LogInformation("Updated battery on robot {name} ", robot.Name);
             }
         }
+
+        private async void OnPoseUpdate(object? sender, MqttReceivedArgs mqttArgs)
+        {
+            var poseStatus = (IsarPoseMessage)mqttArgs.Message;
+            var robot = await _robotService.ReadByName(poseStatus.RobotId);
+            if (robot == null)
+            {
+                _logger.LogWarning(
+                    "Could not find corresponding robot for pose update with ID {id} ",
+                    poseStatus.RobotId
+                );
+            }
+            else
+            {
+                UpdatePoseFromIsarPose(robot.Pose, poseStatus.Pose);
+                await _robotService.Update(robot);
+                _logger.LogInformation("Updated pose on robot {name} ", robot.Name);
+            }
+        }
+        private static void UpdatePoseFromIsarPose(Pose robotPose, IsarPose isarPose)
+        {
+            robotPose.Frame = isarPose.frame.name;
+
+            robotPose.Position.X = isarPose.position.x;
+            robotPose.Position.Y = isarPose.position.y;
+            robotPose.Position.Z = isarPose.position.z;
+
+            robotPose.Orientation.X = isarPose.orientation.x;
+            robotPose.Orientation.Y = isarPose.orientation.y;
+            robotPose.Orientation.Z = isarPose.orientation.z;
+            robotPose.Orientation.W = isarPose.orientation.w;
+        }
```

Comment:
I think this belongs on the IsarPose class as a method `ToRobotPose()` returning a `Pose`, rather than in the MQTT event handler

Reply:
I agree, however we need to send in the `Pose robotPose` argument, since if we let `ToRobotPose()` just return a new pose, then it will be missing the correct id that is mapped in the databae. 
Hence, we need to send in the database robot object and copy the values directly.
We can change this when we achieve saving of pose through JSON serialization

Reply:
Ah, that makes sense, good catch.